pilot/pkg/serviceregistry/kube/controller/controller_test.go

```diff
@@ -192,87 +191,91 @@ func TestServices(t *testing.T) {
 		},
 	})
 
-	ctl, fx := newFakeControllerWithOptions(fakeControllerOptions{networksWatcher: networksWatcher})
-	defer ctl.Stop()
-	t.Parallel()
-	ns := "ns-test"
+	for mode, name := range EndpointModeNames {
+		t.Run(name, func(t *testing.T) {
+			ctl, fx := newFakeControllerWithOptions(fakeControllerOptions{networksWatcher: networksWatcher, mode: mode})
+			defer ctl.Stop()
+			t.Parallel()
```

Comment:
This must be run at the front of a test?

Reply:
I don't think it is required, I preserved what we have been doing before